presto-main/src/main/java/io/prestosql/execution/CommitTask.java

```diff
@@ -37,7 +38,14 @@ public String getName()
     }
 
     @Override
-    public ListenableFuture<?> execute(Commit statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, QueryStateMachine stateMachine, List<Expression> parameters)
+    public ListenableFuture<?> execute(
+            Commit statement,
+            TransactionManager transactionManager,
+            Metadata metadata,
+            AccessControl accessControl,
+            QueryStateMachine stateMachine,
+            List<Expression> parameters,
+            WarningCollector warningCollector)
```

Comment:
Can i suggest separating the addition of the `WarningCollector` to the task interface, from other changes related to table redirection?

Reply:
Sure, makes sense; updated.